pkg/controller/petset/pet.go

```diff
@@ -197,24 +197,28 @@ func (p *apiServerPetClient) Create(pet *pcb) error {
 }
 
 // Update updates the pet in the 'pet' pcb to match the pet in the 'expectedPet' pcb.
+// If the pod object of a pet which to be updated has been changed in server side, we
+// will get the actual value and set pet identity before retries.
 func (p *apiServerPetClient) Update(pet *pcb, expectedPet *pcb) (updateErr error) {
-	var getErr error
 	pc := podClient(p.c, pet.parent.Namespace)
 
-	pod, needsUpdate, err := copyPetID(pet, expectedPet)
-	if err != nil || !needsUpdate {
-		return err
-	}
-	glog.Infof("Resetting pet %v to match PetSet %v spec", pod.Name, pet.parent.Name)
-	for i, p := 0, &pod; ; i++ {
-		_, updateErr = pc.Update(p)
-		if updateErr == nil || i >= updateRetries {
-			return updateErr
+	glog.Infof("Resetting pet %v/%v to match PetSet %v spec", pet.pod.Namespace, pet.pod.Name, pet.parent.Name)
+	for rep := 0; rep < updateRetries; rep++ {
+		updatePod, needsUpdate, err := copyPetID(pet, expectedPet)
+		if err != nil || !needsUpdate {
+			return err
+		}
+		_, updateErr = pc.Update(&updatePod)
+		if updateErr == nil {
```

Comment:
please more the rep < updateRetries exit condition to here like it was before, you're wasting a Get now in the case where we exit the loop